pkg/controllers/provisioning/provisioner.go

```diff
@@ -29,6 +29,7 @@ import (
 var (
 	MaxBatchDuration = time.Second * 10
 	MinBatchDuration = time.Second * 1
+	MaxPodsPerBatch  = 2_000
```

Comment:
Could you add a comment explaining why we're doing this (eg to limit memory consumption)?